src/program.ts

```diff
@@ -3233,6 +3236,21 @@ export class Flow {
     return this.currentFunction.localsByName.get(name);
   }
 
+  getGlobal(expr: ExpressionRef): Global {
+    var exprId = expr.toString();
```

Comment:
Wondering how this is expected to work since each `ExpressionRef` is a unique instruction (unique pointer value to the respective Binaryen `Expression`). Hence, this seems to be limited to the exact same instruction being evaluated multiple times?

Reply:
Yeah. In most cases this true. But if you insert `(<any>console).count(exprId)` after 3240 line and run tests for example you can see sometimes like this:

```
...
5433780: 1
5434332: 1
5434884: 3
5435436: 1
...
```
and this:
```
5421792: 2
5429572: 2
5429688: 2
5429736: 2
5429856: 2
5429904: 2
5430056: 2
5430104: 2
5430224: 2
5430272: 2
5430392: 2
5430440: 2
5430524: 2
5430536: 2
5430584: 2
5430596: 2
5430644: 2
5430656: 2
5430704: 2
5430716: 2
5430832: 2
5430880: 2
5431000: 2
5431048: 2
5431168: 2
5431216: 2
```

Reply:
Remove caching for globals